### PR TITLE
Iteratable batch runner

### PR DESCRIPTION
**Pull request by richfitz**
~Includes #23, so review and merge after that.~

Same basic approach as taken in dfoptim with the fits, to provide a non-blocking interface to the sensitivity. You can access solutions (and anything else really) during the fit. We might want to change the order that we iterate through, but it's not immediately obvious what the best order is, so this just goes from small to large!

### CODE REVIEW

File: src/batch.ts [hunk 1]
```diff
@@ -41,35 +43,49 @@ export class Batch {
      * @param control Optional control parameters to tune the integration
      */
     constructor(run: singleBatchRun, pars: BatchPars, tStart: number, tEnd: number) {
-        this.pars = pars;
+        this.pars = { ...pars, values: [] as number[]};
```
Comment (EmmaLRussell) [inline src/batch.ts:48]: So you're initialising with no values in the pars, and then you'll add in the values as they are used? Could be worth updating the comment to note that

File: src/batch.ts [hunk 2]
```diff
@@ -170,13 +186,21 @@ export interface BatchError {
...
-                         control: Partial<DopriControlParam>): Batch {
+                         control: Partial<DopriControlParam>,
+                         immediate: boolean = true): Batch {
     const run = (p: UserType, t0: number, t1: number) =>
         wodinRun(Model, p, t0, t1, control);
-    return new Batch(run, pars, tStart, tEnd);
+    const ret = new Batch(run, pars, tStart, tEnd);
+    if (immediate) {
+        ret.run();
+    }
```
Comment (EmmaLRussell) [inline src/batch.ts:204]: Is this just to reproduce the current behaviour where blocking run happens immediately?

**Review by EmmaLRussell — APPROVED** [review]
LGTM

File: src/batch.ts (hunk 2 above)
Comment (richfitz) [reply src/batch.ts:204]: yes - if we change the ODE batch version later, we can remove this option

**Pull request merged**